logistics/scripts/modules/settings/settings-management.psm1

```diff
@@ -312,21 +321,75 @@ function New-DevelopmentAppSettings([hashtable] $Settings = @{ }) {
         $developmentSettingsForEngine = (Get-DefaultDevelopmentSettingsByEngine)[$Settings.ApiSettings.Engine]
         $developmentSettingsForEngine = Add-ApplicationNameToConnectionStrings $developmentSettingsForEngine $project
 
-        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine, $Settings
+        $newDevelopmentSettings = Merge-Hashtables $developmentSettingsByProject[$project], $developmentSettingsForEngine
 
         $newDevelopmentSettings = Add-TestHarnessSpecificAppSettings $newDevelopmentSettings $project
 
+        $newDevelopmentSettings = Merge-Hashtables $newDevelopmentSettings, (Get-CredentialSettingsByProject)[$project], $Settings
+
         $projectPath = Get-RepositoryResolvedPath $Project
 
         $newDevelopmentSettingsPath = Join-Path $projectPath "appsettings.development.json"
         New-JsonFile $newDevelopmentSettingsPath $newDevelopmentSettings -Overwrite
         $newSettingsFiles += $newDevelopmentSettingsPath
 
         $credentialSettings = Merge-Hashtables @{ }, (Get-CredentialSettingsByProject)[$project]
-        $newUserSettingsPath = Join-Path $projectPath "appsettings.user.json"
-        New-JsonFile $newUserSettingsPath $credentialSettings
-        $newSettingsFiles += $newUserSettingsPath
+        $userSecrets += ($credentialSettings | ConvertTo-Json -Depth 10 | dotnet user-secrets set --project $projectPath --id (Get-UserSecretsIdByProject).$project)
     }
 
     return $newSettingsFiles
 }
+
+$RegExFilter = "\A(?<curr>[^:]+)(:(?<remain>(?<sub>[^:]+)(:.*)?))?\Z"
+
+function Get-FlattenHash {
+    param([string]$Key,$Value,$Obj)  
+
+    if ($Value) {
+        if ($key -match $RegExFilter) {
+            $Current = $matches.curr
+            if (-not ($Obj | gm $Current)) {
+                        if (-not $Obj.ContainsKey($Current)) { $Obj.$Current = @{ } }
+            }
+
+            if ($matches.remain) {
+                $Obj.$Current = Get-FlattenHash -Key $matches.remain -Value $Value -Obj ($Obj.$Current)
+            } else {
+                $Obj.$Current = $Value
+            }
+            return $Obj
+        }
+    }
+}
```

Comment:
I'd really like to see this end up in logistics\scripts\modules\utility\hashtable.psm1 along with some tests in the hashtable.tests.ps1. This code should probably also be part of that function:
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/5e0943dca2ee00d31bbda22474943f340f73e273/logistics/scripts/modules/settings/settings-management.psm1#L388-L394

I want to be able to send my flat hashtable to this function and get an unflattened one without having to any extra work outside of the function.
As an example you could use a local function like the `Get-FlatHashTable` function:
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/6c1c483433bc5faa0ebb4180e49c640031d6dcde/logistics/scripts/modules/utility/hashtable.psm1#L83-L117

Reply:
I move to hashtable.psm1 file .Thanks